src/ui-test/utils/testUtils.ts

```diff
@@ -161,6 +161,14 @@ export async function closeEditor(title: string, save?: boolean) {
 	const dirty = await new TextEditor().isDirty();
 	await new EditorView().closeEditor(title);
 	if (dirty) {
+		await VSBrowser.instance.driver.wait(() => {
+			try {
+				new ModalDialog();
+				return true;
+			} catch {
+				return false;
+			}
+		});
```

Comment:
better to add some default timeout to avoid infinit waiting failures + also some timeout failure message to better understand what and where happened in case